### PR TITLE
fix vp8 decoder for networking streams.

### PR DESCRIPTION
**Pull request by xuguangxin**
two fixes here:
1. do not treat parser error as fatal error, in network condition, it always have error bitstream, we'd better not treat it as faltal error just skip current frame.
2. skip P frames if we do not see key frame. networking stream many  not start from a key frame, we should skip all P frame until we got a key

### CODE REVIEW

**Comment by xuguangxin** [comment]
Hi @chivakker 
Please help to review.thanks

File: decoder/vaapidecoder_vp8.cpp [hunk 1]
```diff
@@ -542,6 +544,13 @@ YamiStatus VaapiDecoderVP8::decode(VideoDecodeBuffer* buffer)
             status = ensureContext();
             if (status != YAMI_SUCCESS)
                 return status;
+            m_gotKeyFrame = true;
+        }
+        else {
+            if (!m_gotKeyFrame) {
+                //we can't decode frame without key
```
Comment (lizhong1008) [inline decoder/vaapidecoder_vp8.cpp]: Would better add some debug information here, such as WARNING.

Comment (lizhong1008) [reply decoder/vaapidecoder_vp8.cpp]: Seems other codec may have similar issue, do we need to extend m_gotKeyFrame to other codecs?

Comment (xuguangxin) [reply decoder/vaapidecoder_vp8.cpp]: 1. I can add warning,
2. It's not needed by h264, h265. The parser will return error if no sps https://github.com/01org/libyami/blob/apache/codecparsers/h264Parser.cpp#L521
   for vp9, I do not have enough time to fix it, maybe we can file a bug to track it.

**Comment by xuguangxin** [comment]
@lizhong1008 updated and create #637 for vp9

**Comment by xuguangxin** [comment]
@lizhong1008 , is it ok to start test this?

**Comment by xuguangxin** [comment]
Hi @wudping 
Seems lizhong take vocation now, Could you start verification this, you can test h264/h265 flush issue with this in one round

**Comment by wudping** [comment]
okay @xuguangxin , I will do at once.

**Comment by wudping** [comment]
verify + 1

**Comment by lizhong1008** [comment]
Thanks for the update. Review +1

**Pull request merged**